src/category_theory/subobject/limits.lean

```diff
@@ -102,25 +106,62 @@ lemma kernel_subobject_factors_iff {W : C} (h : W ⟶ X) :
   kernel_subobject_arrow_comp, comp_zero],
 kernel_subobject_factors f h⟩
 
+instance is_iso_kernel_subobject_zero_arrow : is_iso (kernel_subobject (0 : X ⟶ Y)).arrow :=
+by { rw ←kernel_subobject_arrow, apply_instance, }
+
+@[simp]
+lemma kernel_subobject_zero {A B : C} : kernel_subobject (0 : A ⟶ B) = ⊤ :=
+eq_of_comm (kernel_subobject_iso _ ≪≫ kernel_zero_iso_source ≪≫ (underlying_iso _).symm)
+  (by simp [kernel_subobject_arrow])
+
 lemma le_kernel_subobject (A : subobject X) (h : A.arrow ≫ f = 0) : A ≤ kernel_subobject f :=
 subobject.le_mk_of_comm (kernel.lift f A.arrow h) (by simp)
 
+/--
+The isomorphism between the kernel of `f ≫ g` and the kernel of `g`,
+when `f` is an isomorphism.
+-/
+def kernel_subobject_iso_comp
+  {X' : C} (f : X' ⟶ X) [is_iso f] (g : X ⟶ Y) [has_kernel g] :
+  (kernel_subobject (f ≫ g) : C) ≅ (kernel_subobject g : C) :=
```

Comment:
Isn't this a direct consequence of `kernel_subobject_comp_mono` below?

Reply:
No, it's the other way round. Here the isomorphism is on the left of the composition, there the mono is on the right of the composition.

Reply:
You are right, my bad. Is there a reason why you stated this as an isomorphism between the representatives instead of an equality of subobjects?

Reply:
They are subobjects of difference objects here.

Reply:
Oh, of course. Sorry for the noise.